docs/pages/installation/reverse-proxy-configuration.rst

```diff
@@ -0,0 +1,169 @@
+Vendors could use this information to enable or disable certain certificates on demand.
+This means that vendors will have full control over which certificate is granted access and which isn't.
+
+Most reverse-proxies require a configured list of CAs to enable mTLS.
+Most of the time, however, they will not be able to configure a list of accepted non-CA certificates.
+Specific certificate acceptance must therefore be done at the application level.
+
+The ``/api/mtls/*`` APIs on :ref:`nuts-registry-api` describe how to obtain an acceptance list in various formats.
+
+Examples
+********
+
+Below are configuration examples on how to configure various reverse proxies.
+
+Apache
+======
+
+Used apache version: 2.4.43
+
+.. code-block:: xml
+
+    <VirtualHost *:443>
+    ServerName proxy-server
+      # activate HTTPS server certificate on the reverse proxy
+        SSLEngine on
+        SSLCertificateFile "<Apache_home>/conf/proxy-server.crt"
+        SSLCertificateKeyFile "<Apache_home>/conf/proxy-server.key"
+        SSLCertificateChainFile "<Apache_home>/conf/proxy-server-ca.crt"
+      # activate the client certificate authentication
+        SSLCACertificateFile "/etc/nuts/data/truststore.pem"
+        SSLVerifyClient require
+        SSLVerifyDepth 1
+        SSLProxyEngine On
+      # initialize the special headers to a blank value to avoid http header forgeries
+        RequestHeader set SSL_CLIENT_CERT ""
+        <Location /api/fhir>
+            # forward certificate for acceptance list checking
+            RequestHeader set SSL_CLIENT_CERT "%{SSL_CLIENT_CERT}s"
+            ProxyPass http://localhost:8080
+            ProxyPassReverse http://localhost:8080
+        </Location>
+    </VirtualHost>
+
+HAProxy
+=======
+
+.. note::
+
+    help requested on a valid HAProxy example
+
+Nginx
+=====
+
+Used nginx version: 1.17.0
+
+.. code-block:: text
+
+    http {
+      # only TLS > 1.2 is acceptable
+      ssl_protocols TLSv1.2 TLSv1.3;
+      ssl_prefer_server_ciphers on;
+
+      access_log /var/log/nginx/access.log;
+      error_log /var/log/nginx/error.log;
+
+      # server on port 80 for HTTP -> HTTPS redirect
+      server {
+        listen 80;
+        server_name example.com;
+        return 301 https://example.com$request_uri;
+      }
+
+      # The HTTPS server, which proxies our requests
+      server {
+        listen 443 ssl;
+        server_name example.com;
+
+        ssl_protocols TLSv1.2 TLSv1.3;
+        # server certificate
+        ssl_certificate /etc/nginx/ssl/example.com/fullchain.pem;
+        ssl_certificate_key /etc/nginx/ssl/example.com/privkey.pem;
+
+        # client certificate,
+        # here we use the exported truststore.pem in the case Nuts is
+        # running on the same machine
+        ssl_trusted_certificate /etc/nuts/data/truststore.pem;
+        # the client certificate is optional, so for example status endpoints can be accessed
```

Comment:
I wouldn't make TLS client verification optional (it also differs from the Apache example).

Do you want "the world" to be able to access status endpoints (and potentially more, in case of a simple bug?). I generally put TLS termination in DMZs for the outside world and access to monitoring/metric/status/administration endpoints is all internal traffic so they are never reachable through the proxies in the DMZ.

I still think it be a good idea to separate these internal endpoints from (semi)public endpoints (like IRMA) by binding them to a different HTTP port. Much safer since you can then actually use your firewalls and just proxy to the public HTTP service. But that requires serious effort in core and all engines so that's probably not for now...

Reply:
I'll make it required and leave out the if-this-then-that problems. Those rise from the setup of internal infra....